Send a message to players nearby a ship taking off/landing

MuckerMayhem opened the pull request:
:cl: Mucker
rscadd: Adds a message that is sent to players in the vicinity of a shuttle taking off or landing.
/:cl:

Send a message to players nearby a ship taking off/landing (aka, players that are on the same z-level as where the ship is landing too or taking off from). This is mostly intended for exoplanets, as given all landing/taking off occurs in small(ish) open area, it'd be pretty easy to see or at least hear ships taking off or landing around you, even if you weren't directly by where said ship is taking off or leaving from.

Cases where you won't get a message:

The ship is taking off from space.
The ship is leaving an area that is space.
The player is on a space turf.

Spookerton commented on code/modules/shuttles/shuttle.dm:
```diff
@@ -102,6 +102,13 @@
+			var/area/A = get_area(start_location)
+
+			for (var/mob/M in GLOB.player_list)
+				if (M.client && !istype(get_turf(M), /turf/space) && !(get_area(M) in src.shuttle_area) && M.z == A.z)
```
```suggestion
				if (M.client && M.z == A.z && !istype(get_turf(M), /turf/space) && !(get_area(M) in shuttle_area))
```
cheaper tests first

Spookerton commented on code/modules/shuttles/shuttle.dm:
```diff
@@ -120,6 +127,13 @@
+						var/area/A = get_area(destination)
+
+						for (var/mob/M in GLOB.player_list)
+							if (M.client && !istype(get_turf(M), /turf/space) && !(get_area(M) in src.shuttle_area) && M.z == A.z)
```
```suggestion
							if (M.client && M.z == A.z && !istype(get_turf(M), /turf/space) && !(get_area(M) in shuttle_area))
```
cheaper tests first

MuckerMayhem committed:
Send a message to players nearby a ship taking off/landing

Adds a message that is seen by all players in the same z-level as a ship is taking off from, or landing in (with some exceptions).
